pkg/asset/installconfig/aws/metadata.go

```diff
@@ -61,15 +60,15 @@ func (m *Metadata) AvailabilityZones(ctx context.Context) ([]string, error) {
 	m.mutex.Lock()
 	defer m.mutex.Unlock()
 
-	if len(m.availabilityZones) == 0 {
-		session, err := m.unlockedSession(ctx)
-		if err != nil {
-			return nil, err
-		}
+	session, err := m.unlockedSession(ctx)
+	if err != nil {
+		return nil, err
+	}
 
+	if len(m.availabilityZones) == 0 {
```

Comment:
We don't need this change. `session` is not used unless `availabilityZones` hasn't been set yet.